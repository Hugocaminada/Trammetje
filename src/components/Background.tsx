import React from 'react'
import {StyleSheet} from 'react-native'
import styled from 'styled-components/native'
import LinearGradient from 'react-native-linear-gradient'

const MainContainer = styled.View`
  flex: 1;
  background-color: red;
`

const BottomHalf = styled.View`
  flex: 1;
  background-color: #e0e0e0;
`
const Content = styled.View`
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
`

interface Props {
  children: any
}

const Background: React.FC<Props> = ({children}) => {
  // const isDarkMode = useColorScheme() === 'dark'

  return (
    <MainContainer>
      <LinearGradient
        colors={[
          '#56CCF2',
          'rgba(187, 107, 217, 0.7)',
          'rgba(187, 107, 217, 0.6)',
        ]}
        style={styles.linearGradient}
      />
      <BottomHalf />
      <Content>{children}</Content>
    </MainContainer>
  )
}

var styles = StyleSheet.create({
  linearGradient: {
    flex: 1,
  },
})

export default Background
