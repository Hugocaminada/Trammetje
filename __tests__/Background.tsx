/**
 * @format
 */

import 'react-native'
import React from 'react'
import {Text} from 'react-native'
import Background from '../src/components/Background'

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer'

test('if it renders correctly', () => {
  const tree = renderer
    .create(
      <Background>
        <Text>Test</Text>
      </Background>,
    )
    .toJSON()
  expect(tree).toMatchSnapshot()
})
