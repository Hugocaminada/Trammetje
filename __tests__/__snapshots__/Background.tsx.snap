// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if it renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "red",
        "flexBasis": 0,
        "flexGrow": 1,
        "flexShrink": 1,
      },
    ]
  }
>
  <View
    colors={
      Array [
        "#56CCF2",
        "rgba(132, 176, 235, 0.8)",
        "rgba(132, 176, 235, 0.8)",
      ]
    }
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "rgba(187, 107, 217, 0.6)",
          "flexBasis": 0,
          "flexGrow": 1,
          "flexShrink": 1,
        },
      ]
    }
  />
  <View
    style={
      Array [
        Object {
          "height": "100%",
          "left": 0,
          "position": "absolute",
          "top": 0,
          "width": "100%",
        },
      ]
    }
  >
    <Text>
      Test
    </Text>
  </View>
</View>
`;
